packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -2323,4 +2323,47 @@ void main() {
     final Offset snackBarTopRight = tester.getTopRight(find.byType(SnackBar));
     expect(snackBarTopRight.dy, 465.0);
   });
+
+    testWidgets('ScaffoldMessengerState clearSnackBars works as expected', (WidgetTester tester) async {
+    const String helloSnackBar = 'Hello SnackBar';
+    const Key tapTarget = Key('tap-target');
+    final GlobalKey<ScaffoldMessengerState> scaffoldMessengerKey = GlobalKey();
+    await tester.pumpWidget(MaterialApp(
+      home: ScaffoldMessenger(
+        key: scaffoldMessengerKey,
+        child: Scaffold(
+          body: Builder(
+            builder: (BuildContext context) {
+              return GestureDetector(
+                key: tapTarget,
+                onTap: () {
+                  ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
+                    content: Text(helloSnackBar),
+                    duration: Duration(seconds: 2),
+                  ));
+                },
+                behavior: HitTestBehavior.opaque,
+                child: Container(
+                  height: 100.0,
+                  width: 100.0,
+                ),
+              );
+            }
+          ),
+        ),
+      ),
+    ));
+    expect(find.text(helloSnackBar), findsNothing);
+    await tester.tap(find.byKey(tapTarget));
+    await tester.tap(find.byKey(tapTarget));
+    await tester.tap(find.byKey(tapTarget));
+    await tester.tap(find.byKey(tapTarget));
+    await tester.tap(find.byKey(tapTarget));
+    expect(find.text(helloSnackBar), findsNothing);
+    await tester.pump(); // schedule animation
+    expect(find.text(helloSnackBar), findsOneWidget);
+    scaffoldMessengerKey.currentState!.clearSnackBars();
+    await tester.pump();
+    expect(find.text(helloSnackBar), findsNothing);
```

Comment:
I am curios why this would result in a SnackBar no longer being visible, if it were completing an exit animation, wouldn't it need more frames than just one pump to fully exit?

Reply:
Hmm, seems curious :thinking: . I've changed it to ```pumpAndSettle``` although it seemed to work earlier as well with just ``` pump```, where technically it shouldn't have